### PR TITLE
EDSC-2927: Panel Redesign

### PR DESCRIPTION
**Pull request by trevorlang**
# Overview

### What is the feature?

Redesigns the panel layout

### What is the Solution?

The site header was moved out of the sidebar and into a global header, adding the functionality to pull in the users first name. The PanelGroup and PanelGroupHeader components were updated to standardize the repeated functionality between the headers of the different routes (title, loading states, breadcrumbs, view, sort, etc). Granule Search has been moved into the sidebar along with the rest of the granule filters.

### What areas of the application does this impact?

 - Global layout (header/sidebar)
 - Search route
 - Project route

# Testing

### Reproduction steps

- **Environment for testing:** Any
- **Collection to test with:** Any

1. Visit the site
2. Navigate to /search, /search/granules, /search/granules/collection-details, /search/granules/granule-details, /projects routes and confirm intended behavior
3. Navigate to all other site pages and confirm unchanged behavior

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/components/Facets/FacetsGroup.scss [hunk 1]
```diff
@@ -63,10 +97,15 @@
...
+    margin-left: 0.5rem;
+    color: darken($color__black--400, 10);
+  }
+
   &__body {
     overflow: hidden;
-    background: $color__black--900;
-    border-bottom: 1px solid $color__black--700;
+    border-bottom: 1px solid darken($color__black--700, 5);
+    // background: $color__black--800;
```
Comment (trevorlang) [inline static/src/js/components/Facets/FacetsGroup.scss:108]: Remove comment

File: static/src/js/components/AppHeader/AppHeader.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+// import PropTypes from 'prop-types'
```
Comment (trevorlang) [inline static/src/js/components/AppHeader/AppHeader.js:2]: Remove comment

File: static/src/js/components/AppHeader/AppHeader.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+const AppHeader = () => (
+  <header className="app-header">
+    <AppLogoContainer />
+    <SecondaryToolbarContainer />
+  </header>
+)
+
+AppHeader.propTypes = {
+  // portal: PropTypes.shape({}).isRequired
```
Comment (trevorlang) [inline static/src/js/components/AppHeader/AppHeader.js:18]: Remove

File: static/src/js/actions/granules.js [hunk 4]
```diff
@@ -585,31 +595,31 @@ export const getProjectGranules = () => (dispatch, getState) => {
...
+export const applyGranuleFilters = granuleFilters => (dispatch, getState) => {
   const state = getState()
 
   // Retrieve data from Redux using selectors
   const focusedCollectionId = getFocusedCollectionId(state)
   const projectCollectionsIds = getProjectCollectionsIds(state)
 
-  // Apply granule filters, ensuring to reset the page number to 1 as this results in a new search
-  dispatch(actions.updateFocusedCollectionGranuleFilters({ pageNum: 1, ...granuleFilters }))
+  if (isEmpty(granuleFilters)) {
```
Comment (trevorlang) [inline static/src/js/actions/granules.js:606]: Double check

File: static/src/js/components/GranuleFilters/__tests__/GranuleFiltersForm.test.js [hunk 5]
```diff
@@ -48,17 +51,104 @@ describe('GranuleFiltersForm component', () => {
...
+
+          expect(props.onUndoExcludeGranule).toHaveBeenCalledTimes(1)
+          expect(props.onUndoExcludeGranule).toHaveBeenCalledWith('COLL_ID')
+        })
+      })
+    })
+  })
+
   describe('Form', () => {
+    test('shows granule searcg by default', () => {
```
Comment (trevorlang) [inline static/src/js/components/GranuleFilters/__tests__/GranuleFiltersForm.test.js:136]: Typo

File: static/src/js/components/MoreActionsDropdown/MoreActionsDropdown.js [hunk 6]
```diff
@@ -27,15 +39,15 @@ export const MoreActionsDropdown = ({
       onClick={e => e.stopPropagation()}
       onKeyPress={e => e.stopPropagation()}
     >
-      <Dropdown className="more-actions-dropdown__dropdown">
+      <Dropdown className="more-actions-dropdown__dropdown dropdown-dark">
```
Comment (trevorlang) [inline static/src/js/components/MoreActionsDropdown/MoreActionsDropdown.js:42]: This is unnecessary

File: static/src/js/components/Panels/PanelGroupHeader.js [hunk 7]
```diff
@@ -1,48 +1,283 @@
...
+                  height: '1.25rem',
+                  width: '100%'
+                }}
+                shapes={titleSkeleton}
+              />
+            )
+            : (
+              <h2 className="panel-group-header__heading">
+                {
+                  secondaryHeading && (
```
Comment (trevorlang) [inline static/src/js/components/Panels/PanelGroupHeader.js:151]: secondaryHeading is no longer a thing

File: static/src/js/components/Panels/PanelGroupHeader.js [hunk 8]
```diff
@@ -1,48 +1,283 @@
...
+                  }
+
+                  return item
+                })
+              }
+            </MoreActionsDropdown>
+          )
+        }
+      </div>
+      {header && header}
```
Comment (trevorlang) [inline static/src/js/components/Panels/PanelGroupHeader.js:223]: No longer needed

File: static/src/js/components/Panels/PanelGroupHeader.js [hunk 9]
```diff
@@ -1,48 +1,283 @@
...
+            {
+              (sortsArray.length > 0 || viewsArray.length > 0) && (
+                <nav className="panel-group-header__heading-meta-secondary">
+                  {
+                    sortsArray.length > 0 && (
+                      <RadioSettingDropdown
+                        id={`panel-group-header-dropdown__sort__${panelGroupId}`}
+                        className="panel-group-header__setting-dropdown"
+                        activeSortOrder={activeSort}
+                        activeIcon={`${activeSort === 'descending' ? 'sort-amount-desc' : 'sort-amount-asc'}`}
```
Comment (trevorlang) [inline static/src/js/components/Panels/PanelGroupHeader.js:261]: No longer needed

**Review by macrouch — APPROVED** [review]

**Review by abbottry — APPROVED** [review]

**Pull request merged**